### PR TITLE
IABot: use a single  variable for all config-related key/values

### PR DESCRIPTION
**Pull request by MusikAnimal**

### CODE REVIEW

File: IABot/deadlink.php [hunk 1]
```diff
@@ -102,43 +105,45 @@
...
 		foreach( $toggleParams[1] as $id=>$variable ) {
-			eval( "if( isset( \$$variable ) ) \$$variable = {$toggleParams[2][$id]};" );
+			eval( "if( isset( \$$config[variable] ) ) \$$config[variable] = {$toggleParams[2][$id]};" );
 		}
 		foreach( $stringParams[1] as $id=>$variable ) {
-			eval( "if( isset( \$$variable ) ) \$$variable = \"{$stringParams[2][$id]}\";" );
-			if( strpos( $variable, "TAGS" ) !== false ) {
-				eval( "if( !empty( \$$variable ) ) \$$variable = explode( ';', \$$variable );
-					   else \$$variable = array();" );
+			eval( "if( isset( \$$config[variable] ) ) \$$config[variable] = \"{$stringParams[2][$id]}\";" );
```
Comment (cyberpower678) [inline IABot/deadlink.php]: Syntax error.

File: IABot/deadlink.php [hunk 2]
```diff
@@ -102,43 +105,45 @@
...
+			if( strpos( $config[variable], "TAGS" ) !== false ) {
+				eval( "if( !empty( \$$config[variable] ) ) \$$config[variable] = explode( ';', \$$config[variable] );
```
Comment (cyberpower678) [inline IABot/deadlink.php]: And here.

File: IABot/deadlink.php [hunk 3]
```diff
@@ -102,43 +105,45 @@
...
+					   else \$$config[variable] = array();" );
```
Comment (cyberpower678) [inline IABot/deadlink.php]: And here.

File: IABot/deadlink.php [hunk 4]
```diff
@@ -102,43 +105,45 @@
...
 			}
 		}
-		
+
 		if( isset( $overrideConfig ) && is_array( $overrideConfig ) ) {
 			foreach( $overrideConfig as $variable=>$value ) {
-				eval( "if( isset( \$$variable ) ) \$$variable = \$value;" );
+				eval( "if( isset( \$$config[variable] ) ) \$$config[variable] = \$value;" );
```
Comment (cyberpower678) [inline IABot/deadlink.php]: And here.

**Comment by cyberpower678** [comment]
Some thoughts.  I think you got all the variables, very impressive, but why are they lowercase now?  The configuration parser, syntax errors aside, will create uppercase keys, which will result the bot in suffering multiple organ failure. :p

Eval won't be needed in this case though, since a variable can assign the key to the array.  Eval was only being used to generate actual variables dynamically.

**Comment by cyberpower678** [comment]
In other words, the eval expressions can be replaced with actual code in the form of "if( isset( $config[$variable] ) stuff...

**Comment by MusikAnimal** [comment]
@cyberpower678 Got rid of the eval, what a nice side effect of using one centralized config variable! :) If we move to a JS file you would by convention use lowercased keys (and I think they should be lowercased anyway), so for now I've just used `strtolower` to convert the current config to lowercase before storing in `$config`. Hopefully this is OK.

**Pull request closed without merging**